src/actions/shared/approve.ts

```diff
@@ -0,0 +1,20 @@
+import { Token, TOKEN_PROGRAM_ID, u64 } from '@solana/spl-token';
+import { Keypair, PublicKey, Transaction } from '@solana/web3.js';
+
+export function createApproveTxs(account: PublicKey, owner: PublicKey, amount: number | u64) {
+  const authority = Keypair.generate();
```

Comment:
In other cases authority can be same for multiple approve tx's please move `authority` to attributes

Reply:
I didn't get it, what you mean